src/agent/backingstore.js

```diff
@@ -0,0 +1,139 @@
+/* Copyright (C) 2020 NooBaa */
+'use strict';
+
+require('../util/dotenv').load();
+
+const fs = require('fs');
+const path = require('path');
+const minimist = require('minimist');
+
+const dbg = require('../util/debug_module')(__filename);
+dbg.set_process_name('backingstore');
+
+const system_store = require('../server/system_services/system_store');
+system_store.get_instance({ standalone: true });
+
+const Agent = require('./agent');
+const fs_utils = require('../util/fs_utils');
+const db_client = require('../util/db_client');
+const nb_native = require('../util/nb_native');
+const json_utils = require('../util/json_utils');
+const auth_server = require('../server/common_services/auth_server');
+
+const HELP = `
+Help:
+
+    "backingstore" is a noobaa-core command runs a local backingstore agent
+    For more information refer to the noobaa docs.
+`;
+
+const USAGE = `
+Usage:
+
+    node src/agent/backingstore <storage-path> [options...]
+`;
+
+const ARGUMENTS = `
+Arguments:
+
+    <storage-path>      Storage dir to use (e.g "data-dir/bucket-name")
+`;
+
+const OPTIONS = `
+Options:
+
+    --port <port>       (required!)                      Listen on port
+    --address <url>     (default wss://localhost:5443)   Connect to the base core address
+    --debug <level>     (default 0)                      Increase debug level
+`;
+
+const WARNINGS = `
+WARNING:
+
+    !!! This feature is WORK IN PROGRESS and can change without notice !!!
+`;
+
+function print_usage() {
+    console.warn(HELP);
+    console.warn(USAGE.trimStart());
+    console.warn(ARGUMENTS.trimStart());
+    console.warn(OPTIONS.trimStart());
+    console.warn(WARNINGS.trimStart());
+    process.exit(1);
+}
+
+async function main(argv = minimist(process.argv.slice(2))) {
+    try {
+        if (argv.help || argv.h) return print_usage();
+        if (argv.debug) {
+            const debug_level = Number(argv.debug) || 5;
```

Comment:
the default when --debug is not even provided will be 0 because we don't get into this `if(argv.debug)` at all.
Only if the number in argv.debug (which in minimist can also happen to be a string or a boolean) is not a number, then it fallsback to 5.
But I think you're right and this is just too confusing so I will change to be clearer.
Thanks